src/plugins/discover/public/application/components/doc_viewer/doc_viewer.tsx

```diff
@@ -35,7 +35,7 @@ export function DocViewer(renderProps: DocViewRenderProps) {
     .getDocViewsSorted(renderProps.hit)
     .map(({ title, render, component }: DocView, idx: number) => {
       return {
-        id: title,
+        id: `kbn_doc_viewer_tab_${idx}`,
```

Comment:
Why was this change introduced? Generally it's better to not use the array position for these kind of identifiers. They will be reused when the array changes which can cause unexpected behavior. If there are uniqueness-concerns we could use both (`kbn_doc_viewer_tab_${idx}_${title}`

Reply:
EUI now uses `id` as a DOM selector and these titles are not valid. We could remove spaces or otherwise transform the title string. Would that be preferable?

Reply:
Makes sense, thanks for the explanation. I checked the code and it's not possible this particular array while change on the fly, so I think it's fine leaving it like this for now - no need to mess around with the title. If it would have been simple it would be preferable to not use the index, but no big deal.